ansible/config_sonic_basedon_testbed.yml

```diff
@@ -80,14 +80,42 @@
     ignore_errors: true
 
   - name: find interface name mapping and individual interface speed if defined from dut
-    port_alias: hwsku="{{ hwsku }}"
+    port_alias:
+      hwsku: "{{ hwsku }}"
+      card_type: "{{ card_type | default('fixed') }}"
+      hostname: "{{ inventory_hostname | default('') }}"
+      start_switchid: "{{ start_switchid | default(0) }}"
     when: deploy is defined and deploy|bool == true
 
   - name: find interface name mapping and individual interface speed if defined with local data
-    port_alias: hwsku="{{ hwsku }}" num_asic="{{ num_asics }}"
+    port_alias:
+      hwsku: "{{ hwsku }}"
+      num_asic: "{{ num_asics }}"
+      card_type: "{{ card_type | default('fixed') }}"
+      hostname: "{{ inventory_hostname | default('') }}"
+      start_switchid: "{{ start_switchid | default(0) }}"
     delegate_to: localhost
     when: deploy is not defined or deploy|bool == false
 
+  - name: find and generate fabric ASIC infomation
+    fabric_info:
+      num_fabric_asic: "{{ num_fabric_asics | default(0) }}"
+      asics_host_pfx: "{{ asics_host_ip | default(None) }}"
+      asics_host_pfx6: "{{ asics_host_ipv6 | default(None) }}"
+
+  - name: set all VoQ system ports information
+    set_fact:
+      all_sysports: "{{ all_sysports | default( [] ) + item.value['ansible_facts']['sysports'] }}"
+    with_dict: "{{ hostvars }}"
+    when: item.value['ansible_facts']['sysports'] is defined
+
+  - name: set all VoQ information for iBGP
+    set_fact:
+      all_inbands: "{{ all_inbands | default( [] ) + [ item.value['voq_inband_ip'] ] }}"
+      all_hostnames: "{{ all_hostnames | default( [] ) + [ item.value['inventory_hostname'] ] }}"
+    with_dict: "{{ hostvars }}"
+    when: item.value['voq_inband_ip'] is defined
+
```

Comment:
Instead of the above logic to loop through all the hosts in the inventory the below code works better using ansible_play_batch as it loops only through the hosts that are in play:

```
  - name: set all VoQ system ports information
    set_fact:
      all_sysports: "{{ all_sysports | default( [] ) + hostvars[item]['sysports'] }}"
    when: hostvars[item]['sysports'] is defined
    loop: "{{ ansible_play_batch }}"

  - name: set all VoQ information for iBGP
    set_fact:
      all_inbands: "{{ all_inbands | default( [] ) + [ hostvars[item]['voq_inband_ip'] ] }}"
      all_hostnames: "{{ all_hostnames | default( [] ) + [ item ] }}"
    when: hostvars[item]['voq_inband_ip'] is defined
    loop: "{{ ansible_play_batch }}"
```